File: smart_travel_FE/lib/data/models/booking/cancellation_policy_response.dart
class CancellationPolicyResponse {
  final bool canCancel;
  final int cancelBeforeHours;
  final String cancelDeadline;
  final double cancellationFeePercent;
  final double estimatedCancellationFee;
  final String message;

  CancellationPolicyResponse({
    required this.canCancel,
    required this.cancelBeforeHours,
    required this.cancelDeadline,
    required this.cancellationFeePercent,
    required this.estimatedCancellationFee,
    required this.message,
  });

  factory CancellationPolicyResponse.fromJson(Map<String, dynamic> json) {
    return CancellationPolicyResponse(
      canCancel: json['canCancel'] ?? false,
      cancelBeforeHours: json['cancelBeforeHours'] ?? 24,
      cancelDeadline: json['cancelDeadline'] ?? '',
      cancellationFeePercent: (json['cancellationFeePercent'] ?? 0).toDouble(),
      estimatedCancellationFee: (json['estimatedCancellationFee'] ?? 0).toDouble(),
      message: json['message'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'canCancel': canCancel,
      'cancelBeforeHours': cancelBeforeHours,
      'cancelDeadline': cancelDeadline,
      'cancellationFeePercent': cancellationFeePercent,
      'estimatedCancellationFee': estimatedCancellationFee,
      'message': message,
    };
  }
}

File: smart_travel_BE/src/main/java/com/example/smart_travel_BE/dto/booking/response/CancellationPolicyResponse.java
package com.example.smart_travel_BE.dto.booking.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CancellationPolicyResponse {
    private boolean canCancel;
    private Integer cancelBeforeHours;          // Số giờ phải hủy trước
    private String cancelDeadline;              // Thời hạn hủy (dạng text)
    private BigDecimal cancellationFeePercent;  // Phí hủy (%)
    private BigDecimal estimatedCancellationFee; // Phí hủy ước tính (nếu hủy bây giờ)
    private String message;                      // Thông báo cho khách
}